client/rpc/manual-xcm/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+use cumulus_primitives_core::ParaId;
+use cumulus_primitives_core::XcmpMessageFormat;
+use futures::{future::BoxFuture, FutureExt as _};
+use jsonrpc_core::Result as RpcResult;
+use jsonrpc_derive::rpc;
+use parity_scale_codec::Encode;
+use xcm::latest::prelude::*;
+
+/// This RPC interface is used to manually submit XCM messages that will be injected into a
+/// parachain-enabled runtime. This allows testing XCM logic in a controlled way in integration
+/// tests.
+#[rpc(server)]
+pub trait ManualXcmApi {
+	/// Inject a downward xcm message - A message that comes from the relay chain.
+	/// You may provide an arbitrary message, or if you provide an emtpy byte array,
+	/// Then a default message (DOT transfer down to ALITH) will be injected
+	#[rpc(name = "xcm_injectDownwardMessage")]
+	fn inject_downward_message(&self, message: Vec<u8>) -> BoxFuture<'static, RpcResult<()>>;
+
+	/// Inject an HRMP message - A message that comes from a dedicated channel to a sibling
+	/// parachain.
+	///
+	/// Cumulus Parachain System seems to have a constraint that at most one hrmp message will be
+	/// sent on a channel per block. At least that's what this comment implies:
+	/// https://github.com/paritytech/cumulus/blob/c308c01b/pallets/parachain-system/src/lib.rs#L204
+	/// Neither this RPC, nor the mock inherent data provider make any attempt to enforce this
+	/// constraint. In fact, violating it may be useful for testing.
```

Comment:
```suggestion
	/// constraint. In fact, violating it may be useful for testing.
	/// The method accepts a paraId as a paremeter, and a bytearray representing an arbitrary message
	/// if you provide an emtpy byte array, then a default message representing the paraId token will be injected
```